### PR TITLE
Bump eventing and knative-operator to v1.7

### PR DESCRIPTION
**Pull request by ReToCode**
## Proposed Changes
- Bump all dependencies to 1.7 in `olm-catalog/serverless-operator/project.yaml` and `hack/update-deps.sh`
- Update dependencies with `make generated-files` and `./hack/update-deps.sh --upgrade`
- Updated the patch `hack/patches/004-disable-manifests-version-match.patch`
- Added another `go mod tidy -compat=1.18`, otherwise go `go mod vendor` did not run
- Updated `go mod tidy` format to `1.18`
- Introduce a new directory and env `GORUN_PATH` variable because of changes in `go_run()` function here: https://github.com/knative/hack/commit/123a2783c4d8e4091e1625e0410d9f60c9a18e1a

As discussed in Slack, I'm unable to update the go dependencies without affecting serving as well. We'll see how this merges together with serving and operator bumps.

### CODE REVIEW

File: olm-catalog/serverless-operator/project.yaml [hunk 1]
```diff
@@ -37,20 +37,20 @@ dependencies:
...
-  eventing: knative-v1.6
+  eventing: knative-v1.7
   # eventing core midstream branch name
-  eventing_artifacts_branch: release-v1.6
+  eventing_artifacts_branch: release-v1.7
 
   eventing_kafka: 1.1.0
   # eventing-kafka-broker promotion tag
-  eventing_kafka_broker: knative-v1.6
+  eventing_kafka_broker: knative-v1.7
```
Comment (matzew) [inline olm-catalog/serverless-operator/project.yaml:46]: I personally would still separate those: eventing, and broker - but let's see

File: openshift-knative-operator/cmd/operator/kodata/knative-eventing/1.7/5-eventing-post-install.yaml [hunk 2]
```diff
@@ -224,5 +224,5 @@ spec:
             runAsNonRoot: true
             capabilities:
               drop:
-              - ALL
+              - all
```
Comment (skonto) [inline openshift-knative-operator/cmd/operator/kodata/knative-eventing/1.7/5-eventing-post-install.yaml:227]: This should be "ALL".

**Comment by skonto** [comment]
```
go: writing stat cache: mkdir /go/pkg/mod/cache/download/gotest.tools: permission denied
```
It shows that:
```
GOPATH=/go
```
Wondering if that has changed compared to previous runs.

File: openshift-knative-operator/cmd/operator/kodata/knative-eventing/1.7/5-eventing-post-install.yaml (hunk 2 above)
Comment (matzew) [reply openshift-knative-operator/cmd/operator/kodata/knative-eventing/1.7/5-eventing-post-install.yaml:227]: yeah, I know, I thought our midstreams have all these fixes - let me double check

Comment (creydr) [reply openshift-knative-operator/cmd/operator/kodata/knative-eventing/1.7/5-eventing-post-install.yaml:227]: https://github.com/openshift-knative/eventing/pull/58

File: openshift/ci-operator/source-image/Dockerfile [hunk 3]
```diff
@@ -6,4 +6,8 @@ COPY --from=registry.ci.openshift.org/openshift/knative-eventing-src:knative-v1.
 COPY --from=registry.ci.openshift.org/openshift/knative-v1.1.0:knative-eventing-kafka-src /go/src/knative.dev/eventing-kafka/ /go/src/knative.dev/eventing-kafka/
 COPY --from=registry.ci.openshift.org/openshift/eventing-kafka-broker-src:knative-v1.7 /go/src/github.com/openshift-knative/eventing-kafka-broker/ /go/src/knative.dev/eventing-kafka-broker/
 
+# Create a temp directory for the go_run() function that is writable by runtime users
```
Comment (matzew) [inline openshift/ci-operator/source-image/Dockerfile:9]: ah, nice

**Review by matzew — APPROVED** [review]
/lgtm
/approve

Feel free to unhold

**Pull request merged**